src/client/menu/qmenu.c

```diff
@@ -223,51 +223,31 @@ extern int keydown[];
 qboolean
 Field_Key(menufield_s *f, int key)
 {
-	switch (key)
-	{
-		case K_KP_SLASH:
-			key = '/';
-			break;
-		case K_KP_MINUS:
-			key = '-';
-			break;
-		case K_KP_PLUS:
-			key = '+';
-			break;
-		case K_KP_HOME:
-			key = '7';
-			break;
-		case K_KP_UPARROW:
-			key = '8';
-			break;
-		case K_KP_PGUP:
-			key = '9';
-			break;
-		case K_KP_LEFTARROW:
-			key = '4';
-			break;
-		case K_KP_5:
-			key = '5';
-			break;
-		case K_KP_RIGHTARROW:
-			key = '6';
-			break;
-		case K_KP_END:
-			key = '1';
-			break;
-		case K_KP_DOWNARROW:
-			key = '2';
-			break;
-		case K_KP_PGDN:
-			key = '3';
-			break;
-		case K_KP_INS:
-			key = '0';
-			break;
-		case K_KP_DEL:
-			key = '.';
-			break;
-	}
+    /*
+     * Ignore keypad in field to prevent duplicate
+     * entries through key presses processed as a
+     * normal char event and additionally as key
+     * event.
+     */
+    switch (key)
+    {
+        case K_KP_SLASH:
+        case K_KP_MINUS:
+        case K_KP_PLUS:
+        case K_KP_HOME:
+        case K_KP_UPARROW:
+        case K_KP_PGUP:
+        case K_KP_LEFTARROW:
+        case K_KP_5:
+        case K_KP_RIGHTARROW:
+        case K_KP_END:
+        case K_KP_DOWNARROW:
+        case K_KP_PGDN:
+        case K_KP_INS:
+        case K_KP_DEL:
+        default:
+            break;
+    }
```

Comment:
The switch-case statement is now superfluous and can be deleted.

Reply:
Deleted.